libsolidity/ast/AST.cpp

```diff
@@ -495,6 +495,24 @@ string Scopable::sourceUnitName() const
 	return *sourceUnit().annotation().path;
 }
 
+bool Declaration::isEnumValue() const
+{
+	solAssert(scope(), "");
+	return dynamic_cast<EnumDefinition const*>(scope());
```

Comment:
Would `return dynamic_cast<EnumValue const*>(this);` be better?

Reply:
I think the current version is more clear (and matches the others)